apps/accounts/automations/change_secret/manager.py

```diff
@@ -180,6 +181,7 @@ def on_host_success(self, host, result):
         while retry_count < max_retries:
             try:
                 recorder.save()
+                account.skip_history_when_saving = self.skip_history_when_saving
                 account.save(update_fields=['secret', 'version', 'date_updated'])
                 break
             except Exception as e:
```

Comment:
This code appears to be for a playbook management system used by Ansible software. There don't seem to be any major anomalies or unnecessary coding here. 

The key changes I'd suggest making:

  1. Use Python's f-string for formatting instead of str.format().
  2. Ensure that `skip_history_when_saving` attribute is not empty.
  3. In case of an error encountered during execution (`Exception`) you can print the error without exiting the loop.

Here's what this might look like with suggested improvements:

```python
def run_playbook(hosts, play_data):
```

I'm also assuming that @@@ and ++ are meant to denote indentation markers (like spaces). The updated version would then look something like:

```python
class ChangeSecretManager(InventoryBasePlaybookManager):
    ...

```